### DISCUSSION TITLE
fn i_slint_core::platform::duration_until_next_timer_update()  return back Some(1s)

### DISCUSSION THREAD

**Discussion by fluolite** [slint-ui/slint#6579] [General]
I have a development board without GPU hardware.
[https://www.rock-chips.com/a/en/products/RK33_Series/2018/0614/907.html](https://www.rock-chips.com/a/en/products/RK33_Series/2018/0614/907.html)

I try to experience its performance with slint-ui, but it has a very low frame rate.

1. when export SLINT_BACKEND=linuxkms-software, fps almost 10fps -> 15fps.
2. when export SLINT_BACKEND=linuxkms-femtovg, fps always 1fps.

I tried to tracking the problem,  and found it that "duration_until_next_timer_update" return value corresponding to fps.
I has a problem.
How is it calculated and waht the factors that can affect it ?

Slint code in app-window.slint:
```
import { Button, VerticalBox, ScrollView } from "std-widgets.slint";

export component AppWindow inherits Window {
    width: 480px;
    height: 800px;

    Rectangle {
        width: 300px;
        height: parent.height;

        Flickable {
            width: parent.width;
            height: parent.height;
            viewport-width: parent.width;
            viewport-height: 100 * 50px;

            for data[index] in 100 : Rectangle {
                x: 0;
                y: index * 50px;
                width: 300px;
                height: 50px;
                background: (Math.mod(index, 2) == 0) ? #00ff00 : #ff0000;

                Text {
                    font-size: 30px;
                    color: #333333;
                    text: index + 1;
                }
            }
        }
    }
}
```

As a contrast, I try it with QT widgets.
When export QT_QPA_PLATFORM=eglfs, the fps almost 1ps.
When export QT_QPA_PLATFORM=linuxfb, the fps more that 30fps, experience is very good.

**Comment by ogoffart** [comment]
Maybe also worth trying the Skia renderer

Note that i_slint_core::platform::duration_until_next_timer_update does NOT contain the animation. So if you are using this function to sleep, you should also check has_active_animation do refresh if there are animations.
See documentation of https://docs.rs/slint/latest/slint/platform/fn.duration_until_next_timer_update.html
